### ISSUE TITLE
Poor error message on erroneous map projection

### ISSUE THREAD

**Issue by fickludd** [neo4j/neo4j#12997]
- Neo4j version: 4.4.11 community
### Steps to reproduce

Run
`WITH [null, 1] AS l RETURN l{p: 2}`

### Expected behavior
User error
`Map projection requires the map variable (l) to be of type map, node or relationship, but was List{NO_VALUE, Long(1)} [1:42]`

### Actual behavior
Neo.DatabaseError.Statement.ExecutionFailed
`List{NO_VALUE, Long(1)} (of class org.neo4j.values.virtual.ListValue$ArrayListValue)`

Cheers guys!

**Comment by mnd999** [comment]
Thanks, we'll take a look at it. Hope all is good with you.

**Comment by fickludd** [comment]
While chipping away at a 60 line cypher query, I hit this problem again today, where the error message was

```
Neo.DatabaseError.Statement.ExecutionFailed

String("a3912") (of class org.neo4j.values.storable.StringWrappingStringValue)
``` 

and the error was getting passing a `String` as the map variable in a Map projection. Reproducible with
``` UNWIND ["s", 2] AS s RETURN s{.prop} ```

**Comment by gem-neo4j** [comment]
Hi! Thanks for reporting, this is now fixed and will be in upcoming releases (5.15 and 4.4.29)
